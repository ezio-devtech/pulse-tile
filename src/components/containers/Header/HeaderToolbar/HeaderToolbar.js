import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import classNames from 'classnames';
import { compose, lifecycle } from 'recompose';

import Sidebar from '../../../presentational/Sidebar/Sidebar';
import toolbarSelector from './selectors';
import { setSidebarVisibility } from '../../../../ducks/set-sidebar-visibility';
import { closeSidebarOnUnmount, openSidebarOnMount } from '../../../../utils/HOCs/sidebar-handle';

const mapDispatchToProps = dispatch => ({ actions: bindActionCreators({ setSidebarVisibility }, dispatch) });

@connect(toolbarSelector, mapDispatchToProps)
@compose(lifecycle(closeSidebarOnUnmount), lifecycle(openSidebarOnMount))
class HeaderToolbar extends PureComponent {
  static propTypes = {
    isSidebarVisible: PropTypes.bool.isRequired,
    name: PropTypes.string.isRequired,
    gpName: PropTypes.string.isRequired,
    gpAddress: PropTypes.string.isRequired,
    dateOfBirth: PropTypes.string.isRequired,
    gender: PropTypes.string.isRequired,
    telephone: PropTypes.string.isRequired,
    userId: PropTypes.string.isRequired,
  };

  static contextTypes = {
    router: PropTypes.shape({
      history: PropTypes.object,
    }),
  };

  state = {
    activeLink: 'patients-summary',
  };

  toggleSidebarVisibility = () => this.props.actions.setSidebarVisibility(!this.props.isSidebarVisible);

  handleGoToState = (state) => {
    this.context.router.history.replace(state);
    this.setState({activeLink: state});
  };

  render() {
    const { isSidebarVisible, name, gpName, gpAddress, dateOfBirth, gender, telephone, userId } = this.props;
    const { activeLink } = this.state;

    return (
      <div>
        <div className="wrap-header-toolbar">
          <div className="container-fluid">
            <div className="header-toolbar">
              <button className={classNames('btn-toggle-sidebar wrap-icon', { 'btn-toggle-sidebar-open': isSidebarVisible })} data-toggle="collapse" data-target="#sidebar-nav" aria-expanded="false" onClick={this.toggleSidebarVisibility}>
                <i className="btn-icon fa fa-bars" />
                <span className="btn-text">Menu</span>
              </button>
              <div className="wrap-patient-info">
                <div className="patient-info-caption">
                  <div className="patient-info-caption-btn btn-dropdown-toggle" />
                  <div className="patient-info-caption-text text-truncate">{name}</div>
                </div>
                <div className="patient-info">
                  <div className="patient-info-group-2">
                    <div className="column-1">
                      <div className="patient-info-item"><span className="key">D.O.B.</span> {dateOfBirth}</div>
                      <div className="patient-info-item"><span className="key">Phone:</span> {telephone}</div>
                    </div>
                    <div className="column-2">
                      <div className="patient-info-item"><span className="key">Gender:</span> {gender}</div>
                      <div className="patient-info-item"><span className="key">NHS No.</span> <span>{userId}</span></div>
                    </div>
                  </div>
                  <div className="patient-info-group-1">
                    <div className="patient-info-item significant hidden-xs">{name}</div>
                    <div className="patient-info-item"><span className="key">Doctor:</span> {gpName}</div>
                  </div>
                  <div className="patient-info-item"><span className="key">Address:</span> {gpAddress}</div>
                </div>
              </div>
            </div>
          </div>
        </div>
        {isSidebarVisible && <Sidebar
          userId={userId}
          goToState={this.handleGoToState}
          activeLink={activeLink}
        />}
      </div>
    )
  }
}

export default HeaderToolbar;
