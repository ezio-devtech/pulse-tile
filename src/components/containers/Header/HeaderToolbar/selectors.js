import _ from 'lodash/fp';
import { createSelector } from 'reselect';

import { getDDMMMYYYY } from '../../../../utils/time-helpers.utils';

const patientToolbarSelector = createSelector(
  ({ patientsSummaries }) => patientsSummaries,
  (state, props) => _.getOr(null, 'match.params.userId', props),
  (patientsSummaries, userId) => {
    const patientSummary = _.getOr({}, userId, patientsSummaries);
    const { name = '', gpName = '', gpAddress = '', dateOfBirth = '', gender = '', telephone = '' } = patientSummary;
    return ({
      name,
      gpName,
      gpAddress,
      gender,
      telephone, //TODO should be preformatted
      userId,
      dateOfBirth: getDDMMMYYYY(dateOfBirth) });
  });

export default patientToolbarSelector;
