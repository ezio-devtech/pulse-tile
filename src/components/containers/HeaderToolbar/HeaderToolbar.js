import React from 'react';

const HeaderToolbar = props => <div>

</div>;

export default HeaderToolbar;
