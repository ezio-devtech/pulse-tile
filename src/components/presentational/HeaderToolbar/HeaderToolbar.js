import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { push } from 'react-router-redux'

import routersSelector from './selectors';
import { mainPagesTitles } from '../../../config/client-urls.constants'

const mapDispatchToProps = dispatch => ({ actions: bindActionCreators({ push }, dispatch) });

@connect(routersSelector, mapDispatchToProps)
class HeaderToolbar extends PureComponent {
  static propTypes = {
    router: PropTypes.shape().isRequired,
  };

  render() {
    const { router } = this.props;
    const routerHash = (router.location.hash.split('?')[0]).split('#')[1];

    const routerHeaderToolbar = (state) => {
      const headerToolbar = mainPagesTitles[state].headerTitle;

      return headerToolbar;
    };

    return (
      <div className="wrap-header-title">
        <div className="container-fluid">
          <div className="header-title">{routerHeaderToolbar(routerHash)}</div>
        </div>
      </div>
    )
  }
}

export default HeaderToolbar;
