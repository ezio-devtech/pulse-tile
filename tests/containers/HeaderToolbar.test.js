import React from 'react';
import configureStore from 'redux-mock-store';
import sinon from 'sinon'
import Enzyme, { render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux'
import { StaticRouter } from 'react-router'

import HeaderToolbar from '../../src/components/containers/Header/HeaderToolbar/HeaderToolbar';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore();
const dispatch = sinon.spy();
const store = mockStore({
  runtime: {},
  router: {
    location: {
      pathname: '/',
      search: '',
      hash: '#/patients/9999999000/procedures',
    },
  },
  isSidebarVisible: true,
});
const context = {};

describe('Component <HeaderToolbar />', () => {
  it('should renders correctly', () => {
    const headerToolbar = render(
      <Provider store={store}>
        <StaticRouter location="someLocation" context={context}>
          <HeaderToolbar
            dispatch={dispatch}
          />
        </StaticRouter>
      </Provider>)
    expect(headerToolbar).toMatchSnapshot();
  });
});

