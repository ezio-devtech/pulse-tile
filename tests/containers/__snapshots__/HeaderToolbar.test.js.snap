// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <HeaderToolbar /> should renders correctly 1`] = `
<div>
  <div
    class="wrap-header-toolbar"
  >
    <div
      class="container-fluid"
    >
      <div
        class="header-toolbar"
      >
        <button
          aria-expanded="false"
          class="btn-toggle-sidebar wrap-icon btn-toggle-sidebar-open"
          data-target="#sidebar-nav"
          data-toggle="collapse"
        >
          <i
            class="btn-icon fa fa-bars"
          >
            
          </i>
          <span
            class="btn-text"
          >
            Menu
          </span>
        </button>
        <div
          class="wrap-patient-info"
        >
          <div
            class="patient-info-caption"
          >
            <div
              class="patient-info-caption-btn btn-dropdown-toggle"
            >
              
            </div>
            <div
              class="patient-info-caption-text text-truncate"
            >
              
            </div>
          </div>
          <div
            class="patient-info"
          >
            <div
              class="patient-info-group-2"
            >
              <div
                class="column-1"
              >
                <div
                  class="patient-info-item"
                >
                  <span
                    class="key"
                  >
                    D.O.B.
                  </span>
                   
                </div>
                <div
                  class="patient-info-item"
                >
                  <span
                    class="key"
                  >
                    Phone:
                  </span>
                   
                </div>
              </div>
              <div
                class="column-2"
              >
                <div
                  class="patient-info-item"
                >
                  <span
                    class="key"
                  >
                    Gender:
                  </span>
                   
                </div>
                <div
                  class="patient-info-item"
                >
                  <span
                    class="key"
                  >
                    NHS No.
                  </span>
                   
                  <span>
                    
                  </span>
                </div>
              </div>
            </div>
            <div
              class="patient-info-group-1"
            >
              <div
                class="patient-info-item significant hidden-xs"
              >
                
              </div>
              <div
                class="patient-info-item"
              >
                <span
                  class="key"
                >
                  Doctor:
                </span>
                 
              </div>
            </div>
            <div
              class="patient-info-item"
            >
              <span
                class="key"
              >
                Address:
              </span>
               
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="sidebar-underlay showSidebar"
    >
      
    </div>
    <div
      class="sidebar showSidebar"
      role="navigation"
    >
      <div
        class="sidebar-nav"
      >
        <div>
          <ul
            class="sidebar-nav-list"
          >
            <li
              class="sidebar-nav-item"
            >
              <a
                class="sidebar-nav-link"
                href="/patients/null/patients-summary"
              >
                Patient Summary
              </a>
            </li>
            <li
              class="sidebar-nav-item"
            >
              <a
                class="sidebar-nav-link"
                href="/patients/null/diagnoses"
              >
                Problems / Diagnosis
              </a>
            </li>
            <li
              class="sidebar-nav-item"
            >
              <a
                class="sidebar-nav-link"
                href="/patients/null/medications"
              >
                Medications
              </a>
            </li>
            <li
              class="sidebar-nav-item"
            >
              <a
                class="sidebar-nav-link"
                href="/patients/null/allergies"
              >
                Allergies
              </a>
            </li>
            <li
              class="sidebar-nav-item"
            >
              <a
                class="sidebar-nav-link"
                href="/patients/null/contacts"
              >
                Contacts
              </a>
            </li>
            <li
              class="sidebar-nav-item"
            >
              <a
                class="sidebar-nav-link"
                href="/patients/null/clinicalNotes"
              >
                Clinical Notes
              </a>
            </li>
            <li
              class="sidebar-nav-item"
            >
              <a
                class="sidebar-nav-link"
                href="/patients/null/vaccinations"
              >
                Vaccinations
              </a>
            </li>
            <li
              class="sidebar-nav-item"
            >
              <a
                class="sidebar-nav-link"
                href="/patients/null/personalNotes"
              >
                Personal Notes
              </a>
            </li>
            <li
              class="sidebar-nav-item"
            >
              <a
                class="sidebar-nav-link"
                href="/patients/null/results"
              >
                Test Results
              </a>
            </li>
            <li
              class="sidebar-nav-item"
            >
              <a
                class="sidebar-nav-link active"
                href="/patients/null/procedures"
              >
                Procedures
              </a>
            </li>
            <li
              class="sidebar-nav-item"
            >
              <a
                class="sidebar-nav-link"
                href="/patients/null/events"
              >
                Events
              </a>
            </li>
            <li
              class="sidebar-nav-item"
            >
              <a
                class="sidebar-nav-link"
                href="/patients/null/referrals"
              >
                Referrals
              </a>
            </li>
            <li
              class="sidebar-nav-item"
            >
              <a
                class="sidebar-nav-link"
                href="/patients/null/mdt"
              >
                MDT
              </a>
            </li>
            <li
              class="sidebar-nav-item"
            >
              <a
                class="sidebar-nav-link"
                href="/patients/null/orders"
              >
                Orders
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
